### PR TITLE
Add API for fetching packuments

### PR DESCRIPTION
**Pull request by andreypopp**
**WIP: DO NOT MERGE YET BUT PLEASE PROVIDE FEEDBACK**

As discussed on package.community Discord I've added an API for fetching packuments.

Currently:
* Registry fetcher fetches the actual packument.
* All other fetchers just return `null` which indicates that `finalizePackument` routine should fetch the corresponding manifest for the package spec and synthesise a packument out of it.

Tests are failing for me for some reason (can't really see what's wrong yet). If you are fine with the approach proposed — I'll fix the test suite and add more tests specifically for the packument API.

It's better to review a commit by commit.

### CODE REVIEW

File: lib/fetchers/registry/manifest.js [hunk 1]
```diff
@@ -47,7 +36,7 @@ function getManifest (uri, registry, spec, opts) {
         )
         opts.preferOffline = false
         opts.preferOnline = true
-        return fetchPackument(uri, spec, registry, opts).then(packument => {
+        return fetchPackument(uri, egistry, spec, opts).then(packument => {
```
Comment (zkat) [inline lib/fetchers/registry/manifest.js]: quick note that there's a typo here

Comment (andreypopp) [reply lib/fetchers/registry/manifest.js]: Embarrassing! Fixed.

**Comment by zkat** [comment]
This seems like a good approach to me! Sorry for taking so long to respond. I looked through and I think this is something I'd take, assuming tests are there 👍

**Comment by andreypopp** [comment]
Thanks! Totally np about response time — it was a weekend, time to rest not
code :) I'll add tests then.

**Comment by zkat** [comment]
@andreypopp I'm gonna write up some tests myself on this flight, so I'll take this PR home for you. I'm at a point where I could really use the packument API on something :)

**Comment by zkat** [comment]
replaced by #157

**Pull request closed without merging**